src/niclips/image/_convert.py

```diff
@@ -62,53 +59,49 @@ def normalize(
     vmax: float | None = None,
 ) -> np.ndarray:
     """Normalize data using vmin/vmax if provided, otherwise data min/max."""
-    data = np.asarray(data)
-    if vmin is None:
-        vmin = np.nanmin(data)
-    if vmax is None:
-        vmax = np.nanmax(data)
-    data = (data - vmin) / max(vmax - vmin, EPS)
-    data = np.clip(data, 0, 1)
-    return data
+    vmin = np.nanmin(data) if vmin is None else vmin
+    vmax = np.nanmax(data) if vmax is None else vmax
+    scale = vmax - vmin
+    if scale > EPS:
+        return np.clip((data - vmin) / scale, 0, 1)
+    return np.zeros_like(data)
 
 
 def scale(
```

Comment:
We might also need something like this one to deal with non-isotropic voxels:

```python
def scale_pixel_units(
    img: Image.Image,
    pixdim: tuple[float, float],
    target_res: float | None = None,
    resample: Image.Resampling = Image.Resampling.NEAREST,
) -> Image.Image:
    imw, imh = img.size
    pixw, pixh = pixdim
    width = pixw * imw
    height = pixh * imh
    target_res = target_res or min(pixw, pixh)
    new_imw = round((pixw * imw) / target_res)
    new_imh = round((pixh * imw) / target_res)
    new_img = img.resize((new_imw, new_imh), resample=resample)
    return new_img
```

Reply:
Made a few changes in [1b5c654](https://github.com/childmindresearch/niftyone/pull/34/commits/1b5c654d5a542b5ca0021ab048cbeb06963c6491): 

- Re-added an `to_iso` function. This calls on `nibabel.processing.resample_to_output` to do the processing and only works on 3D niftis; this shouldn't be an issue given even in 4D cases, we are always extracting out a single 3D volume at a time. This also ensures that the voxels are always the same dimension regardless of which slice is taken.
- Fixed the scaling function to set to targeted height and corresponding width. If the provided height is not even, a warning is provided to the user, while made to be an even dimension.